File: remove_duplicate.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        p = 0 
        q = 1
        while q < len(nums):
            if nums[q] != nums[p]:
                nums[p+1] = nums[q]
                p += 1
            q += 1
        return p + 1

File: remove_duplicate.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length,k = 0;
        for(int i=0;i<n;i++){
            int j = i;
            while(j<n && nums[i]==nums[j]){
                j++;
            }
            nums[k++] = nums[i];
            i = j - 1;
        }
        return k;
    }
}